install/kubernetes/tetragon: Helm chart add serviceAnnotations and serviceLabels

yieon-lyon opened the pull request:
charts: add serviceAnnotations and serviceLabels for additive Service metadata management

Tetragon Helm chart currently exposes only serviceLabelsOverride which forces full replacement of the Service labels and makes day to day label management error prone.

This patch fixes this by adding:

A new values key serviceLabels as an empty map in values.yaml to allow appending labels without losing chart defaults.

A new values key serviceAnnotations as an empty map in values.yaml to allow appending annotations without losing chart defaults.

Service template logic that preserves compatibility. If serviceLabels is set it is merged with existing chart labels or serviceLabelsOverride to produce the final label set. If serviceAnnotations is set it is merged with existing chart annotations to produce the final annotations set.

Example values:
```
serviceLabels: {}
# serviceLabels:
#   k8s-app: tetragon

serviceAnnotations: {}
# serviceAnnotations:
#  prometheus.io/port: "2112"
#  prometheus.io/scrape: "true"
```

As a secondary note, these additive fields can simplify label based selection in systems such as Prometheus.

<!--
Thanks for contributing! Please ensure your pull request adheres to the following guidelines:

- [ ] All commits contain a well written commit message and are signed-off (see [Submitting a pull request](https://tetragon.io/docs/contribution-guide/submitting-a-pull-request/)).
- [ ] All code is covered by unit and/or end-to-end tests where feasible.
- [ ] All generated files are updated if needed (see [Making changes](https://tetragon.io/docs/contribution-guide/making-changes/)).
- [ ] Provide a title or release-note blurb suitable for the release notes (see [guidelines](https://tetragon.io/docs/contribution-guide/release-notes/#release-note-blurb-in-pr)).
- [ ] Update documentation and write an upgrade note if needed (see [guidelines](https://tetragon.io/docs/contribution-guide/release-notes/#upgrade-notes)).
- [ ] Are you a user of Tetragon? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md) in the Cilium repository.
-->

mtardy reviewed (commented):
Hello 👋 I'm really sorry it looks like this PR was overlooked. I think it makes sense. Could you take a look at the CI errors and fix your patch, you might need to run `make -C install/kubernetes` and maybe shorten your commit title (use `make checkpatch` to see, might need a rebase to get that).

yieon-lyon committed:
install/kubernetes/tetragon: Helm chart add service Annotations and Labels

charts: add serviceAnnotations and serviceLabels for additive Service metadata management

Tetragon Helm chart currently exposes only serviceLabelsOverride which forces full replacement of the Service labels and makes day to day label management error prone.

This patch fixes this by adding:

A new values key serviceLabels as an empty map in values.yaml to allow appending labels without losing chart defaults.

A new values key serviceAnnotations as an empty map in values.yaml to allow appending annotations without losing chart defaults.

Service template logic that preserves compatibility. If serviceLabels is set it is merged with existing chart labels or serviceLabelsOverride to produce the final label set. If serviceAnnotations is set it is merged with existing chart annotations to produce the final annotations set.

As a secondary note, these additive fields can simplify label based selection in systems such as Prometheus.

Signed-off-by: Lyon <yieon1230@gmail.com>

mtardy reviewed (approved):
oh sorry it slipped again, you could have re-requested for review for me to see it. Let me see if we can still merge it.

mtardy commented:
thanks for the fix, sorry for the handling of this PR @yieon-lyon
